indexer/categories_holder.hpp

```diff
@@ -71,8 +70,8 @@ class CategoriesHolder
   template <class ToDo>
   void ForEachCategory(ToDo && toDo) const
   {
-    for (IteratorT i = m_type2cat.begin(); i != m_type2cat.end(); ++i)
-      toDo(*i->second);
+    for (auto & p : m_type2cat)
```

Comment:
Is it possible to use const ref here? Otherwise it's not clear - may toDo modify m_type2cat contents or not.